src/ray/core_worker/reference_count.cc

```diff
@@ -766,6 +766,7 @@ void ReferenceCounter::MergeRemoteBorrowers(const ObjectID &object_id,
 void ReferenceCounter::CleanupBorrowersOnRefRemoved(
     const ReferenceTable &new_borrower_refs, const ObjectID &object_id,
     const rpc::WorkerAddress &borrower_addr) {
+  absl::MutexLock lock(&mutex_);
```

Comment:
Before you merge the PR, let's run the micro benchmark  to make sure there is no regression.

Reply:
Where do you see the performance could be impacted? Is it because we need to lock/unlock more often as we have locks for both subscriber and here?

Reply:
Whenever when you have lock it might (or might not) impact the performance, and running microbenchmark is the easiest way to make sure it's not causing any trouble.